src/main.dev.ts

```diff
@@ -426,19 +433,47 @@ app.on('window-all-closed', () => {
   }
 });
 
-app
-  .whenReady()
-  .then(createWindow)
-  .catch((err: Error) => {
-    logger.error(`createWindow error ${err}`);
-    handleFatalError(err);
-  })
-  .then(() => {
-    return updater?.checkForUpdates();
-  })
-  .catch((err: Error) => {
-    logger.error(`Auto update error ${err}`);
-  });
+if (!isOSSupported) {
+  app
+    .whenReady()
+    .then(() => {
+      // eslint-disable-next-line promise/no-nesting
+      dialog
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        .showMessageBox(undefined, {
+          type: 'error',
+          buttons: ['Okay'],
+          title: 'ExpressLRS Configurator',
+          detail: 'Unsupported OS Version Detected',
+          message: ``,
+        })
+        .then(() => {
+          process.exit(1);
+        })
+        .catch((dialogErr) => {
+          logger.error('failed to show error dialog', dialogErr.stack);
+          process.exit(1);
+        });
+    })
+    .catch((err: Error) => {
+      logger.error(`Unsupported OS dialog error ${err}`);
+    });
+} else {
+  app
+    .whenReady()
+    .then(createWindow)
```

Comment:
I think we should move the alert code over here. Create a new anonymous function, if the os is unsupported stop the execution with the alert, otherwise return `createWindow();` promise and proceed as usual.